nodes/image_processing.py

```diff
@@ -316,6 +317,146 @@ def execute(
+                "interpolation": (["bilinear", "nearest", "bicubic", "area"],),
+                "antialias": (
+                    "BOOLEAN",
+                    {"default": True},
+                ),
+            },
+        }
+
+    RETURN_TYPES = (
+        "IMAGE",
+        "MASK",
+    )
+    FUNCTION = "execute"
+    CATEGORY = IMAGE_PROCESSING_CAT
+    DESCRIPTION = """Resize images based on total pixel count (megapixels) while keeping proportions.
+
+    Instead of specifying exact width and height, this node lets you target a specific image size in megapixels. For example:
+    - 1 megapixel = 1000x1000 pixels for a square image
+    - 0.5 megapixels = roughly 707x707 pixels
+    - 2 megapixels = roughly 1414x1414 pixels
+
+    This is useful when you want to:
+    - Resize a batch of images to a consistent size/quality level
+    - Reduce memory usage by targeting a specific megapixel count
+    - Maintain image proportions while hitting a target file size
+    - Prepare images for specific platforms with megapixel limits
+
+    The image will keep its original proportions - a wide image will stay wide, just with the total pixel count you specify."""
+
+    def get_dimensions(self, megapixels: float, original_width: int = 0, original_height: int = 0) -> tuple[int, int]:
+        """Calculate target dimensions based on megapixels while preserving aspect ratio.
+
+        Args:
+            megapixels (float): Target size in megapixels
+            original_width (int): Original image width (optional)
+            original_height (int): Original image height (optional)
+
+        Returns:
+            tuple[int, int]: Target (width, height)
+        """
+        total_pixels = megapixels * 1000000
+
+        if original_width > 0 and original_height > 0:
+            # Preserve aspect ratio if original dimensions are provided
+            aspect_ratio = original_width / original_height
+            height = math.sqrt(total_pixels / aspect_ratio)
+            width = height * aspect_ratio
+        else:
+            # Default to square dimensions if no original dimensions
+            width = height = math.sqrt(total_pixels)
+
+        return (round(width), round(height))
+
+    def execute(
+        self,
+        megapixels: float = 1.0,
+        image: Optional[torch.Tensor] = None,
+        mask: Optional[torch.Tensor] = None,
+        interpolation: str = "bilinear",
+        antialias: bool = True,
+    ) -> tuple[torch.Tensor, torch.Tensor]:
+        if not isinstance(image, torch.Tensor) and not isinstance(mask, torch.Tensor):
+            raise ValueError("Either image or mask must be provided")
+
+        # Get original dimensions from input tensors if available
+        if isinstance(image, torch.Tensor):
+            _, orig_height, orig_width, _ = image.shape
+        elif isinstance(mask, torch.Tensor):
+            _, orig_height, orig_width, _ = mask.shape
+        else:
```

Comment:
I would not add this else branch here. I found it a bit confusing to read, until I understood that this will never be reached because of the check above. I suggest to remove the check above and add the `ValueError` here.